### PR TITLE
Fix broken default build on macOS xcode

### PR DESCRIPTION
**Pull request by kcgen**
# Description

Default builds on macOS fail when xcode's Clang doesn't support the `-march=native` flag:

```
meson setup build
meson compile -C build
```

```text
[1/352] Compiling C object subprojects/zlib-ng/libzlib.a.p/compress.c.o
FAILED: subprojects/zlib-ng/libzlib.a.p/compress.c.o 
clang: error: the clang compiler does not support '-march=native'
```

Note that this doesn't affect the CI builds, which limit the SIMD instruction sets to NEON (on Arm) and SSE2+SSSE3 (on x86-64).

# Manual testing

Tested default builds on macOS, Linux, and Raspberry Pi (Debian Buster).

# Checklist

_Please tick the items as you have addressed them. Don't remove items; leave the ones that are not applicable unchecked._

I have:

- [x] followed the project's [contributing guidelines](https://github.com/dosbox-staging/dosbox-staging/blob/master/CONTRIBUTING.md) and [code of conduct](https://github.com/dosbox-staging/dosbox-staging/blob/master/CODE_OF_CONDUCT.md).
- [x] performed a self-review of my code.
- [x] commented on the particularly hard-to-understand areas of my code.
- [x] split my work into well-defined, bisectable commits, and I [named my commits well](https://github.com/dosbox-staging/dosbox-staging/blob/main/CONTRIBUTING.md#commit-messages).
- [x] applied the appropriate labels (bug, enhancement, refactoring, documentation, etc.)
- [x] [checked](https://github.com/dosbox-staging/dosbox-staging/blob/main/scripts/compile_commits.sh) that all my commits can be built.
- [x] confirmed that my code does not cause performance regressions (e.g., by running the Quake benchmark).
- [ ] added unit tests where applicable to prove the correctness of my code and to avoid future regressions.
- [ ] made corresponding changes to the documentation or the website according to the [documentation guidelines](https://github.com/dosbox-staging/dosbox-staging/blob/main/DOCUMENTATION.md).
- [ ] [locally verified](https://github.com/dosbox-staging/dosbox-staging/blob/main/DOCUMENTATION.md#previewing-documentation-changes-locally) my website or documentation changes.

### CODE REVIEW

File: meson.build [hunk 1]
```diff
@@ -243,6 +243,118 @@ if os_family_name == 'BSD'
...
+        ssse3_test = cc.run(
+            ssse3_test_code,
+            args: ssse3_cflag,
+            name: 'SSSE3 instruction set test',
+        )
+        if (ssse3_test.compiled() and ssse3_test.returncode() == 0)
+            zlib_ng_options += ['ssse3']
+            extra_flags += ssse3_cflag
+            simd_summary = 'SSE2 and SSSE3'
+            message('Enabling the SSSE3 instruction set')
```
Comment (johnnovak) [inline meson.build:352]: Huh, this all seems rather complicated; isn't there a simpler way?

Comment (kcgen) [reply meson.build:352]: Unfortunately not; the compiler will happily generate the instructions but execution on the target is the only way test them. These are tiny zero-dependency builds, so they fly by during the setup phase (and aren't tested again for all subsequent builds).

Comment (johnnovak) [reply meson.build:352]: Ok, I'm far from being a build automation expert, so I'll trust you on this 😄 Seemed a bit like a maintenance burden (perhaps) but if you're saying it's necessary, I'm fine with it.

File: meson.build [hunk 2]
```diff
@@ -243,6 +243,118 @@ if os_family_name == 'BSD'
...
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+# If the user asked for 'native' SIMD flags then we check if
+# their compiler supports the native arch type. If it doesn't,
+# we try to enable NEON (on Arm) and SSE2 and/or SSSE3 (on x86).
+#
+# Currently zlib-ng is the driving force behind these checks as
+# it provides a meaningful speedup in video and frame capture.
+#
+# Note that these build flags needs to be defined before adding
+# subpackages because the flags are used when compiling them.
```
Comment (johnnovak) [inline meson.build:256]: Nice explanatory comments 🏅

**Review by johnnovak — APPROVED** [review]
👍🏻

**Pull request merged**